### ISSUE TITLE
Problem with original request

### ISSUE THREAD

**Issue by sagikazarmark** [thephpleague/route#7]
I am using @alexbilbie's Proton which uses this package for routing internally. My problem is that the original request object is not passed to the controller. When the dispatcher is called (with data originating from an already existing Request object), it creates a new Request which does not contain anything (not query params, etc).

I opened this issue here because I think it is rather an issue with routing than an issue with Proton itself. However, this still might occur because of improper use of router, so if this is a bug in Proton, I need some help to fix it.

**Comment by sagikazarmark** [comment]
For complete report:

I am running the latest proton (already using league/route) with php built-in server on PHP 5.5

**Comment by philipobenito** [comment]
The request object is fetched from the container so needs to be built within the container. I can take a look at Proton at some point or if @alexbilbie can take a look to ensure Proton is registering the request with the container?

Something like:

```php
$container->add('Symfony\Component\HttpFoundation\Request', function () {
    return Symfony\Component\HttpFoundation\Request::buildFromGlobals();
});
```

In the meantime @sagikazarmark you could register it yourself within Proton by adding this somewhere within your bootstrap.

```php
$app->getContainer()->add('Symfony\Component\HttpFoundation\Request', function () {
    return Symfony\Component\HttpFoundation\Request::buildFromGlobals();
});
```

**Comment by sagikazarmark** [comment]
@philipobenito thanks for your response. I am going to open a PR in Proton.

**Comment by philipobenito** [comment]
No problem, sorry for the delay

**Comment by jakefolio** [comment]
I had a similar issue with just working off of the http://route.thephpleague.com/getting-started/. Every time I tried to access a query param, there would be nothing.

I followed the instructions you gave here and added it to the container, and everything worked good.

This seems like a necessary step in accessing query params when using route. I'd either reopen this as a bug or update the docs on the getting started.

**Comment by philipobenito** [comment]
This is certainly something I can add for the 1.x branches that depend on Symfony\HttpFoundation, but v2 will actually depend on PSR-7 rather than a specific implementation with the documentation heavily padded out with some example implementations and how to build the requests within the app, along with v2 will come a skeleton web application (codename Sterling) and a skeleton restful API (codename Lana) that will both handle that for the user in the bootstrap and also provide a good example for anybody who wants to roll their own.

**Comment by philipobenito** [comment]
Will be addressed by #26
